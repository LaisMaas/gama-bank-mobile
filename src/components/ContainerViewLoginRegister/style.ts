import {Dimensions} from 'react-native';
import styled from "styled-components/native";

export const ContainerView = styled.View`
  height: ${Dimensions.get("window").height};
  flex: 1;
  align-items: center;
  justify-content: center;
`;