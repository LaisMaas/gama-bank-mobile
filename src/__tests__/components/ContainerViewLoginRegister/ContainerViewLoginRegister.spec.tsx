import React from 'react';
import renderer from 'react-test-renderer';
import ContainerViewLoginRegister from '../../../components/ContainerViewLoginRegister';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native")
    return {
      ...actualNav,
      useFocusEffect: () => jest.fn(),
      useNavigation: () => ({
        navigate: jest.fn(),
      }),
    }
  })

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

describe('<ContainerViewLoginRegister />', () => {
  it('Should render <ContainerViewLoginRegister />', () => {
    const tree = renderer.create(<ContainerViewLoginRegister/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});