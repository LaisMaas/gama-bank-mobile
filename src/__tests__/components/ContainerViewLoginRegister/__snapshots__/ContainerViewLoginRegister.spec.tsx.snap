// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerViewLoginRegister /> Should render <ContainerViewLoginRegister /> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "flexWrap": "nowrap",
        "justifyContent": "space-between",
        "paddingBottom": 30,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      },
    ]
  }
/>
`;
